pkg/datapath/linux/devices.go

```diff
@@ -278,15 +288,139 @@ func (dm *DeviceManager) updateDevicesFromRoutes(l3DevOK bool, routes []netlink.
 
 		viable := dm.isViableDevice(l3DevOK, info.hasDefaultRoute, link)
 		if viable {
+			log.WithField(logfields.Device, name).Info("Added new device")
```

Comment:
I think that in an ideal world, we would probably log this _after_ reacting to the new device rather than before, so that the reader in the logs could say "Cilium now knows about this device and has reacted to it". I guess we could make this clearer, eg by saying "New device detected, evaluating which functionality to apply" or similar. This way if we end up debugging some issue that's related to device detection, then we will know how to interpret the log compared to when the functionality is successfully configured on this new device.